tools/clang/unittests/HLSLExec/LongVectors.cpp

```diff
@@ -692,608 +504,989 @@ void fillShaderBufferFromLongVectorData(std::vector<BYTE> &ShaderBuffer,
   return;
 }
 
-// Returns the compiler options string to be used for the shader compilation.
-// Reference ShaderOpArith.xml and the 'LongVectorOp' shader source to see how
-// the defines are used in the shader code.
-template <typename T>
-std::string TestConfig<T>::getCompilerOptionsString() const {
+template <typename T, typename OUT_TYPE, size_t ARITY, typename OP_TYPE>
+std::string getCompilerOptionsString(OP_TYPE OpType, size_t VectorSize,
+                                     uint16_t ScalarInputFlags,
+                                     std::string ExtraDefines) {
+  OpTypeMetaData<OP_TYPE> OpTypeMetaData = getOpTypeMetaData(OpType);
 
-  std::stringstream CompilerOptions("");
+  std::stringstream CompilerOptions;
 
   if (is16BitType<T>())
     CompilerOptions << " -enable-16bit-types";
 
-  CompilerOptions << " -DTYPE=" << getHLSLInputTypeString();
-  CompilerOptions << " -DNUM=" << LengthToTest;
+  CompilerOptions << " -DTYPE=" << getHLSLTypeString<T>();
+  CompilerOptions << " -DNUM=" << VectorSize;
 
   CompilerOptions << " -DOPERATOR=";
-  if (Operator)
-    CompilerOptions << *Operator;
+  if (OpTypeMetaData.Operator)
+    CompilerOptions << *OpTypeMetaData.Operator;
 
   CompilerOptions << " -DFUNC=";
-  if (Intrinsic)
-    CompilerOptions << *Intrinsic;
+  if (OpTypeMetaData.Intrinsic)
+    CompilerOptions << *OpTypeMetaData.Intrinsic;
 
-  // For most of the ops this string is std::nullopt.
-  if (SpecialDefines)
-    CompilerOptions << " " << *SpecialDefines;
+  CompilerOptions << " " << ExtraDefines;
 
-  CompilerOptions << " -DOUT_TYPE=" << getHLSLOutputTypeString();
+  CompilerOptions << " -DOUT_TYPE=" << getHLSLTypeString<OUT_TYPE>();
 
-  CompilerOptions << " -DBASIC_OP_TYPE=" << getBasicOpTypeHexString();
+  CompilerOptions << " -DBASIC_OP_TYPE=0x" << std::hex << ARITY;
```

Comment:
nit: -DBASIC_OP_TYPE would be better named as -DOP_ARITY now IMO. I can also change that in a subsequent change of mine.

Reply:
For this change I don't want to actually modify the shaders at all, so renaming defines for them is out of scope.